### PR TITLE
replaced dp-s3 with dp-s3/v2

### PR DESCRIPTION
**Pull request by BarathaAberathne**
### What

* replace dps3 version into v2
* functions replaced to match the new version in store.go

### How to review

- Check if the all build tests are passed

### Who can review

@Anyone

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -2,6 +2,8 @@ module github.com/ONSdigital/dp-dataset-exporter
 
 go 1.22
 
+replace github.com/ONSdigital/dp-s3 => github.com/ONSdigital/dp-s3/v2 v2.1.0
```
Comment (lindenmckenzie) [inline go.mod]: You don't need a replace statement for this. You need to replace all the code references and then change the go.mod file to require dp-s3/v2

**Review by lindenmckenzie — CHANGES_REQUESTED** [review]

File: file/store.go [hunk 2]
```diff
@@ -50,12 +50,14 @@ func NewStore(
 	localstackHost string,
 ) (*Store, error) {
 
-	uploader, err := s3client.NewUploader(region, publicBucket)
+	//uploader, err := s3client.NewUploader(region, publicBucket)
```
Comment (lindenmckenzie) [inline file/store.go]: Why is this commented out?

File: go.mod [hunk 3]
```diff
@@ -32,34 +25,29 @@ require (
 	github.com/ONSdigital/dp-net/v2 v2.11.2
 )
 
+require golang.org/x/sync v0.7.0 // indirect
```
Comment (lindenmckenzie) [inline go.mod]: Why is this in it's own block?

File: go.mod [hunk 4]
```diff
@@ -32,34 +25,29 @@ require (
...
+
 require (
+	github.com/ONSdigital/dp-s3/v2 v2.1.0
```
Comment (lindenmckenzie) [inline go.mod]: Why is this in the indirect dependencies block?

**Review by lindenmckenzie — CHANGES_REQUESTED** [review]
Would suggest running `go mod tidy` to tidy your go mod.

File: go.mod (hunk 3 above)
Comment (BarathaAberathne) [reply go.mod]: This is added automatically when running go get & go mod tidy.

File: go.mod (hunk 4 above)
Comment (BarathaAberathne) [reply go.mod]: This is added automatically when running go get & go mod tidy.

File: go.mod [hunk 5]
```diff
@@ -1,65 +1,53 @@
 module github.com/ONSdigital/dp-dataset-exporter
 
-go 1.22
+go 1.23
+
+toolchain go1.23.2
```
Comment (lindenmckenzie) [inline go.mod]: Don't want this.

File: go.mod [hunk 6]
```diff
@@ -1,65 +1,53 @@
...
 	github.com/smartystreets/goconvey v1.8.1
-	go.opentelemetry.io/contrib/instrumentation/github.com/gorilla/mux/otelmux v0.53.0
-	go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp v0.53.0
-	golang.org/x/time v0.5.0 // indirect
+	go.opentelemetry.io/contrib/instrumentation/github.com/gorilla/mux/otelmux v0.57.0
+	go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp v0.57.0
 )
 
 require (
 	github.com/ONSdigital/dp-api-clients-go v1.43.0
```
Comment (lindenmckenzie) [inline go.mod]: Why are we direct importing two versions of `dp-api-clients-go`?

File: go.mod [hunk 7]
```diff
@@ -1,65 +1,53 @@
...
 	github.com/ONSdigital/dp-kafka/v4 v4.1.0
-	github.com/ONSdigital/dp-net/v2 v2.11.2
+	github.com/ONSdigital/dp-net/v2 v2.12.0
+	github.com/ONSdigital/dp-s3/v2 v2.1.0
```
Comment (lindenmckenzie) [inline go.mod]: These should all be in the top block.

**Review by lindenmckenzie — CHANGES_REQUESTED** [review]

**Review by lindenmckenzie — APPROVED** [review]

**Pull request merged**